### PR TITLE
add -mhz command line option, correct open bus read behavior

### PR DESCRIPTION
**Pull request by mooinglemur**
image

For the open bus reads, rather than `$C0` as David originally suspected, it's actually the high byte of the target address, most likely floating from the instruction fetch from the last clock.  This change reflects that behavior in the emu.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -514,6 +516,11 @@ usage()
 	printf("\tInstall the second VIA chip expansion at $9F10\n");
 	printf("-testbench\n");
 	printf("\tHeadless mode for unit testing with an external test runner\n");
+	printf("-mhz <integer>\n");
+	printf("\tRun the emulator with a system clock speed other than the default of\n");
+	printf("\t8 MHz. Valid values are in the range of 1-40, inclusive. This option\n");
```
Comment (irmen) [inline src/main.c:521]: this seems fine to me as it is right now, I mean the default is 8 so 1 mhz is already at 1/8th speed. But in general, do we perhaps want a float for this?  So that we can go even lower , into the Kiloherz range?

File: src/main.c [hunk 2]
```diff
@@ -956,6 +963,18 @@ main(int argc, char **argv)
 			argv++;
 			testbench=true;
 			headless=true;
+		} else if (!strcmp(argv[0], "-mhz")){
+			argc--;
+			argv++;
+			if (!argc || argv[0][0] == '-') {
+				usage();
+			}
+			MHZ = (int)strtol(argv[0], NULL, 10);
```
Comment (irmen) [inline src/main.c:972]: cast to uint8_t instead?

File: src/memory.c [hunk 3]
```diff
@@ -168,7 +168,7 @@ real_read6502(uint16_t address, bool debugOn, uint8_t bank)
 		if (ramBank < num_ram_banks) {
 			return RAM[0xa000 + (ramBank << 13) + address - 0xa000];
 		} else {
-			return OPEN_BUS_READ;
+			return (address >> 8) & 0xff; // open bus read
```
Comment (irmen) [inline src/memory.c:171]: I think I would have split this into a separate commit to not conflate this unrelated change with the command line option change. Or is this needed because of the MHZ change?

File: src/main.c (hunk 1 above)
Comment (mooinglemur) [reply src/main.c:521]: There's some integer math on MHZ in audio.c, but other than that it otherwise could possibly behave like a float.

File: src/main.c (hunk 2 above)
Comment (mooinglemur) [reply src/main.c:972]: probably better

File: src/memory.c (hunk 3 above)
Comment (mooinglemur) [reply src/memory.c:171]: This project has never followed a strict practice needing a separate commit or PR for trivial changes.  I don't think it's worth splitting this one out, but I'll keep it in mind so reverts are easier.

File: src/main.c (hunk 1 above)
Comment (irmen) [reply src/main.c:521]: I wonder who ever is going to use that though, <1 mhz... could always change it if someone actually requests it

File: src/memory.c (hunk 3 above)
Comment (irmen) [reply src/memory.c:171]: I agree the change seems trivial but I wasn't sure if it was  somehow related to the MHZ change somewhere down the line

**Review by irmen — APPROVED** [review]
